purchase_request/models/stock_rule.py

```diff
@@ -124,4 +126,22 @@ def create_purchase_request(self, procurement_group):
                 pr.write({"origin": procurement.origin})
         # Create Line
         request_line_data = rule._prepare_purchase_request_line(pr, procurement)
+        # check if request has lines for same product and data
+        # if yes, update qty instead of creating new line
+        same_product_date_request_line = pr.line_ids.search(
+            [
+                ("product_id.id", "=", request_line_data["product_id"]),
+                ("date_required", "=", request_line_data["date_required"]),
+            ],
+            limit=1,
+        )
+        if same_product_date_request_line:
+            new_product_qty = (
+                same_product_date_request_line.product_qty
+                + request_line_data["product_qty"]
+            )
+            same_product_date_request_line.write({"product_qty": new_product_qty})
+        else:
+            # Create Line
+            purchase_request_line_model.create(request_line_data)
```

Comment:
the same line will be created again right after, no?

Reply:
ah yes, i forgot to remove line #147, will do it with commit update.

Reply:
yes needs fixing too.